client-fs.go

```diff
@@ -85,6 +86,102 @@ func (f *fsClient) GetURL() clientURL {
 	return *f.PathURL
 }
 
+func (f *fsClient) Watch(recursive bool) (*watchObject, *probe.Error) {
+	eventChan := make(chan Event)
+	errorChan := make(chan *probe.Error)
+	doneChan := make(chan bool)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, probe.NewError(err)
+	}
+
+	go func() {
+		<-doneChan
+
+		watcher.Close()
+		close(eventChan)
+		close(errorChan)
+	}()
+
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+
+				switch {
+				case event.Op&fsnotify.Create == fsnotify.Create:
+					if i, err := os.Stat(event.Name); err != nil {
+						errorChan <- probe.NewError(err)
+					} else if !i.IsDir() {
+						// we want files
+					} else if !recursive {
```

Comment:
Wouldn't it make sense to not watch at all? - do not understand the logic here btw. Can you explain?